athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/manager/JdbcMetadataHandler.java

```diff
@@ -258,11 +266,16 @@ private ResultSet getColumns(final String catalogName, final TableName tableHand
             throws SQLException
     {
         String escape = metadata.getSearchStringEscape();
-        return metadata.getColumns(
-                catalogName,
-                escapeNamePattern(tableHandle.getSchemaName(), escape),
-                escapeNamePattern(tableHandle.getTableName(), escape),
-                null);
+        String schema = escapeNamePattern(tableHandle.getSchemaName(), escape);
+        String table = escapeNamePattern(tableHandle.getTableName(), escape);
+        String databaseproductname = metadata.getDatabaseProductName();
+        
+        if (databaseproductname != null && !databaseproductname.isEmpty() && databaseproductname.equals("HDB")) {
+                   schema = schema.toUpperCase();
+                   table = table.toUpperCase();
```

Comment:
Similar questions to above, why do we need these specual cases?

Reply:
Schema and table name are in uppercase in HANA and HANA JDBC driver expects them in right case. Athena console for example was passing Schema and table name in lower case (even though actuals are in uppercase) so need to convert the case. Here i am checking if database is HANA (HDB - internal name) and then change the case

Reply:
Would be good if we can handle this in the HANA specific code, not int he JdbcMetadataHandler. you might need to add a method that the HANA specific code implements but Id' rather not have hard coded cases here like HDB

Reply:
Does not look like that's possible. As getSchema function in same class (JdbcMetadataHandler.java) calls getColumns and based on metadata builds the resultset. Beside "HDB" is name of the product driven from metadata as well which will remain same across all SAP HANA releases/versions.

Reply:
Done.